vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -1,370 +1,376 @@
-          } else {
-            clearPageCallbacks([String(rangeMin)]);
-          }
-          comboBox.dataProvider(params, callback);
-        } else if (rangeMax - rangeMin + 1 !== activePages.length) {
-          // Wasn't a sequential page index, clear the cache so combo-box will request for new pages
-          clearPageCallbacks();
-        } else {
-          // The requested page was sequential, extend the requested range
-          const startIndex = params.pageSize * rangeMin;
-          const endIndex = params.pageSize * (rangeMax + 1);
-
-          if (!this._debouncer || !this._debouncer.isActive()) {
-            serverFacade.requestData(startIndex, endIndex, params);
-          } else {
-            this._debouncer = Debouncer.debounce(
-              this._debouncer,
-              timeOut.after(200),
-              () => serverFacade.requestData(startIndex, endIndex, params));
-          }
-        }
-      }
-    }
-
-    comboBox.$connector.filter = function (item, filter) {
-      filter = filter ? filter.toString().toLowerCase() : '';
-      return comboBox._getItemLabel(item).toString().toLowerCase().indexOf(filter) > -1;
-    }
-
-    comboBox.$connector.set = function (index, items, filter) {
-      if (filter != serverFacade.getLastFilterSentToServer()) {
-        return;
-      }
-
-      if (index % comboBox.pageSize != 0) {
-        throw 'Got new data to index ' + index + ' which is not aligned with the page size of ' + comboBox.pageSize;
-      }
-
-      if (index === 0 && items.length === 0 && pageCallbacks[0]) {
-        // Makes sure that the dataProvider callback is called even when server
-        // returns empty data set (no items match the filter).
-        cache[0] = [];
-        return;
-      }
-
-      const firstPageToSet = index / comboBox.pageSize;
-      const updatedPageCount = Math.ceil(items.length / comboBox.pageSize);
-
-      for (let i = 0; i < updatedPageCount; i++) {
-        let page = firstPageToSet + i;
-        let slice = items.slice(i * comboBox.pageSize, (i + 1) * comboBox.pageSize);
-
-        cache[page] = slice;
-      }
+(function () {
+    const tryCatchWrapper = function (callback) {
+        return window.Vaadin.Flow.tryCatchWrapper(callback, 'Vaadin Combo Box', 'vaadin-combo-box-flow');
     };
 
-    comboBox.$connector.updateData = function (items) {
-      // IE11 doesn't work with the transpiled version of the forEach.
-      for (let i = 0; i < items.length; i++) {
-        let item = items[i];
-
-        for (let j = 0; j < comboBox.filteredItems.length; j++) {
-          if (comboBox.filteredItems[j].key === item.key) {
-            comboBox.set('filteredItems.' + j, item);
-            break;
-          }
-        }
-      }
-    }
+    // Not using ES6 imports in this file yet because the connector in V14 must
+    // still work in Legacy bower projects. See: `ironListConnector-es6.js` for
+    // the Polymer3 approach.
```

Comment:
This comment should be moved back to top of file outside the IIFE but also this should say `comboBoxConnector-es6.js` instead of `ironListConnector-es6.js` 😃

Reply:
Updated.